src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -51,6 +51,12 @@ public async Task Invoke(HttpContext context)
 
             // IMPORTANT: Changes to authorization logic should be mirrored in MVC's AuthorizeFilter
             var authorizeData = endpoint?.Metadata.GetOrderedMetadata<IAuthorizeData>() ?? Array.Empty<IAuthorizeData>();
```

Comment:
@rynowak \ @JamesNK can we have `GetOrderedMetadata` return an `IReadOnlyList<T>`? It internally caches an `T[]` so there isn't any code change involved with it.

Reply:
@Eilon, @rynowak pointed out that this would be a breaking change since the API exists in 2.2. However, this seems like a worthwhile change to make - other APIs and users are likely to benefit from this change. How do you feel about this change?

Reply:
We could just do `Count()` here. It'll effectively be the same: https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Count.cs#L12-L22